### PR TITLE
Dev 037 starring products

### PR DESCRIPTION
**Pull request by Paul-Gerarts**
Don't forget to test the new functionalities!

### CODE REVIEW

File: src/test/java/be/syntra/devshop/DevshopBack/controllers/ProductControllerTest.java [hunk 1]
```diff
@@ -296,4 +307,88 @@ void canUpdateCategoryTest() throws Exception {
...
+        final StarRatingDto ratingDto = createRatingDto();
+        when(ratingService.getRatingFromUser(dummyProduct.getId(), rating.getUserName())).thenReturn(rating);
+        when(starRatingMapper.mapToDto(rating)).thenReturn(ratingDto);
+
+        // when
+        ResultActions resultActions =
+                mockMvc.perform(
+                        get("/products/" + rating.getUserName() + "/ratings/" + dummyProduct.getId())
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(jsonUtils.asJsonString(ratingDto)));
```
Comment (lenshuygh) [inline src/test/java/be/syntra/devshop/DevshopBack/controllers/ProductControllerTest.java:326]: adding payload to the call isn't needed, only thing the method expects is the id

File: src/test/java/be/syntra/devshop/DevshopBack/controllers/ProductControllerTest.java [hunk 2]
```diff
@@ -296,4 +307,88 @@ void canUpdateCategoryTest() throws Exception {
...
+        final StarRatingSet ratingsDto = new StarRatingSet(ratings);
+        when(productService.getAllRatingsFromProduct(dummyProduct.getId())).thenReturn(ratings);
+        when(starRatingMapper.mapToStarRatingSet(ratings)).thenReturn(ratingsDto);
+
+        // when
+        ResultActions resultActions =
+                mockMvc.perform(
+                        get("/products/ratings/" + dummyProduct.getId())
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(jsonUtils.asJsonString(ratingsDto)));
```
Comment (lenshuygh) [inline src/test/java/be/syntra/devshop/DevshopBack/controllers/ProductControllerTest.java:380]: same as above

File: src/main/java/be/syntra/devshop/DevshopBack/services/ProductServiceImpl.java [hunk 3]
```diff
@@ -171,4 +175,32 @@ public Product findById(Long id) {
...
+        productRepository.save(product);
+        return product;
+    }
+
+    private Set<StarRating> update(Set<StarRating> ratings, StarRating rating) {
+        Set<StarRating> allRatings = new HashSet<>();
+        allRatings.addAll(ratings);
+        allRatings.remove(rating);
+        allRatings.add(rating);
+        return allRatings;
```
[the remark this answers is not in the document]
Comment (Paul-Gerarts) [reply src/main/java/be/syntra/devshop/DevshopBack/services/ProductServiceImpl.java:204]: as discussed; we both can't really think of anything better?

File: src/test/java/be/syntra/devshop/DevshopBack/controllers/ProductControllerTest.java (hunk 1 above)
Comment (Paul-Gerarts) [reply src/test/java/be/syntra/devshop/DevshopBack/controllers/ProductControllerTest.java:326]: fixed

File: src/test/java/be/syntra/devshop/DevshopBack/controllers/ProductControllerTest.java (hunk 2 above)
Comment (Paul-Gerarts) [reply src/test/java/be/syntra/devshop/DevshopBack/controllers/ProductControllerTest.java:380]: fixed

**Pull request merged**